### DISCUSSION TITLE
Help

### DISCUSSION THREAD

**Discussion by kebab03** [Bodmer/TFT_eSPI#1027] [Q&A - Fonts] [answered]
Good morning, can you please tell me what is the necessity of <User_Setups.h>  in <User_Setups_Select.h>if you give the option for every controller such as <User_Setups/Setup3_ILI9163.h>?
thank you in advance .

**Comment by Bodmer** [comment, marked as answer]
This is down to evolution of the library and to provide backwards compatibility. Originally when the[ library was created ](https://github.com/Bodmer/TFT_ILI9341_ESP)it had a "User_Setup.h" file only. As more displays, processors and features were added the User_Setup_Select.h was added to make changing setups very fast and simple.

Edit: [corrected link](https://github.com/Bodmer/TFT_ILI9341_ESP).
